feat: Action bar refactor to match fiori 3

manu-kr opened the pull request:
## Related Issue

https://github.com/SAP/fundamental-styles/issues/565
## Description

added properties and values to match Fiori 3.
documentation updated.
test scenarios simplified.

## Screenshots
> **NOTE:** If you've made any style changes, please provide appropriate screenshots (before and after) to help reviewers.
>
> To see examples of which screenshots to include, go to [Screenshot Examples](https://github.com/SAP/fundamental-styles/wiki/Pull-Request-Screenshot-Examples).

### Before:
![image](https://user-images.githubusercontent.com/53512176/73822172-87547f00-481b-11ea-8ab6-d107d8b3bca1.png)


### After:
![image](https://user-images.githubusercontent.com/53512176/73822278-b4a12d00-481b-11ea-9d2d-24e3ed20adeb.png)




after npm run test:update
![image](https://user-images.githubusercontent.com/53512176/73822587-6fc9c600-481c-11ea-9d56-c30f88312d67.png)

droshev commented:
@manu-kr I see that you added 2 images for the nested list. Is this intentional?

manu-kr commented:
> @manu-kr I see that you added 2 images for the nested list. Is this intentional?

>@droshev I will check on this.

manu-kr commented on src/action-bar.scss:
```diff
@@ -72,42 +91,46 @@ $block: #{$fd-namespace}-action-bar;
 
   &__description {
     @include fd-reset();
     @include fd-var-color("color", $fd-action-bar-description-color, --fd-color-text-3);
```
Sure i will remove this.

manu-kr commented:
> > @manu-kr I see that you added 2 images for the nested list. Is this intentional?
> 
> > @droshev I will check on this.

These files are committed unintentionally and changes are reverted and committed back previous version'

jbadan commented on src/action-bar.scss:
```diff
@@ -43,25 +41,47 @@ $block: #{$fd-namespace}-action-bar;
+  padding-top: 0.5rem;
+  padding-bottom: 0.5rem;
   display: flex;
+  display: -webkit-flex;
```
PostCSS should be handling this for you - we shouldn't need to add webkit fallbacks manually.

manu-kr replied:
removed this.

jbadan commented on src/action-bar.scss:
```diff
@@ -72,42 +92,49 @@ $block: #{$fd-namespace}-action-bar;
-    margin-bottom: fd-space(base);
+    font-size: var(--sapFontHeader3Size);
+    font-family: var(--sapFontHeaderFamily);
+    font-weight: normal;
```
Shouldn't the font-weight be in the reset?

jbadan replied:
See https://github.com/SAP/fundamental-styles/blob/master/src/mixins/_mixins.scss#L10

manu-kr replied:
Yes you're right. Fixed this by removing font-weight:normal;

stefanoScalzo commented on src/action-bar.scss:
```diff
@@ -43,25 +41,47 @@ $block: #{$fd-namespace}-action-bar;
   }
 
+  @include fd-screen(s) {
+    padding-left: map_get($fd-action-bar-padding-x, "s");
```
where are you getting these measurements

manu-kr replied:
All these measurements based on the fiori 3 standards and here is the new design prepared based on the fiori 3 specs like typography spacings etc.
https://sapariba.invisionapp.com/share/7WSNCMKJQHP#/screens/370170777_ActionBar

manu-kr commented:
> please make sure that all discussions are addressed

Sure, I have addressed  all the issue.

droshev commented:
@manu-kr The PR looks good to me. Can you add examples on the documentation page with cozy buttons?

InnaAtanasova commented:
In the visuals that you provided a link to, the description is shown to go till the end of the container (under the buttons). 
<img width="1132" alt="Screen Shot 2020-02-18 at 3 06 06 PM" src="https://user-images.githubusercontent.com/39598672/74773761-87fc0500-5260-11ea-9b6f-2bf3f00695d9.png">

In the implementation the description container has the same width as the header container and after that truncates. 
<img width="1418" alt="Screen Shot 2020-02-18 at 2 54 45 PM" src="https://user-images.githubusercontent.com/39598672/74773923-cee9fa80-5260-11ea-9600-9081593a1b91.png">

I didn't see any rules about the truncation of the description. Are you sure it needs to be truncated? Either way the "container" should go bellow the actions and then eventually truncate.

InnaAtanasova commented:
There's problem with the title alignment:
<img width="1292" alt="Screen Shot 2020-02-18 at 3 04 47 PM" src="https://user-images.githubusercontent.com/39598672/74774325-79621d80-5261-11ea-81ee-01b3da50af98.png">
<img width="1164" alt="Screen Shot 2020-02-18 at 3 04 52 PM" src="https://user-images.githubusercontent.com/39598672/74774326-79fab400-5261-11ea-964d-1155fcbbd458.png">

InnaAtanasova commented:
<img width="371" alt="Screen Shot 2020-02-18 at 3 17 11 PM" src="https://user-images.githubusercontent.com/39598672/74774458-c47c3080-5261-11ea-99d1-d3848c2244ed.png">

The mobile example should have 1rem padding left and right

InnaAtanasova commented:
I think someone from your team mentioned that you need the title to be editable??? Do you still need this functionality?

<img width="1057" alt="Screen Shot 2020-02-18 at 3 27 39 PM" src="https://user-images.githubusercontent.com/39598672/74775232-46b92480-5263-11ea-897f-cb8a1b302e1b.png">

InnaAtanasova commented:
Not sure the dimensions should stay the same for Mobile. Right now you are using buttons in compact mode and not the cosy mode. For a11y requirements the buttons in mobile should be cosy, which will prob affect the height of the action bar. Could you please check this with the designers?

<img width="1318" alt="Screen Shot 2020-02-18 at 3 30 39 PM" src="https://user-images.githubusercontent.com/39598672/74775401-a1eb1700-5263-11ea-800a-41181391c0d9.png">

InnaAtanasova commented on src/action-bar.scss:
```diff
@@ -43,25 +41,47 @@ $block: #{$fd-namespace}-action-bar;
 
   &__header {
+    flex-grow: 1;
+    width: 75%;
```
this code should be after the reset function

manu-kr replied:
Placed after reset

InnaAtanasova commented:
Instead of using media queries you can add modifier classes for the responsive behavior and let the user add these classes based on their needs. You can check the Page responsive behavior https://sap.github.io/fundamental-styles/layouts/page-layout.html

manu-kr commented:
> In the visuals that you provided a link to, the description is shown to go till the end of the container (under the buttons).
> <img alt="Screen Shot 2020-02-18 at 3 06 06 PM" width="1132" src="https://user-images.githubusercontent.com/39598672/74773761-87fc0500-5260-11ea-9b6f-2bf3f00695d9.png">
> 
> In the implementation the description container has the same width as the header container and after that truncates.
> <img alt="Screen Shot 2020-02-18 at 2 54 45 PM" width="1418" src="https://user-images.githubusercontent.com/39598672/74773923-cee9fa80-5260-11ea-9600-9081593a1b91.png">
> 
> I didn't see any rules about the truncation of the description. Are you sure it needs to be truncated? Either way the "container" should go bellow the actions and then eventually truncate.

@InnaAtanasova There is requirement to truncate, description as well.I have modified the structure as per the visual spec

manu-kr commented:
> I think someone from your team mentioned that you need the title to be editable??? Do you still need this functionality?
> 
> <img alt="Screen Shot 2020-02-18 at 3 27 39 PM" width="1057" src="https://user-images.githubusercontent.com/39598672/74775232-46b92480-5263-11ea-897f-cb8a1b302e1b.png">

@InnaAtanasova 
Yes , there is a requirement we are going to consider this in upcoming sprint.

manu-kr commented:
> Not sure the dimensions should stay the same for Mobile. Right now you are using buttons in compact mode and not the cosy mode. For a11y requirements the buttons in mobile should be cosy, which will prob affect the height of the action bar. Could you please check this with the designers?
> 
> <img alt="Screen Shot 2020-02-18 at 3 30 39 PM" width="1318" src="https://user-images.githubusercontent.com/39598672/74775401-a1eb1700-5263-11ea-800a-41181391c0d9.png">

Added example with cozy button for mobile example

manu-kr commented:
> There's problem with the title alignment. The text is not centered vertically.
> <img alt="Screen Shot 2020-02-18 at 3 04 47 PM" width="1292" src="https://user-images.githubusercontent.com/39598672/74774325-79621d80-5261-11ea-81ee-01b3da50af98.png">
> <img alt="Screen Shot 2020-02-18 at 3 04 52 PM" width="1164" src="https://user-images.githubusercontent.com/39598672/74774326-79fab400-5261-11ea-964d-1155fcbbd458.png">

@InnaAtanasova ,

FIxed. and now its aligned properly. 

> There's problem with the title alignment. The text is not centered vertically.
> <img alt="Screen Shot 2020-02-18 at 3 04 47 PM" width="1292" src="https://user-images.githubusercontent.com/39598672/74774325-79621d80-5261-11ea-81ee-01b3da50af98.png">
> <img alt="Screen Shot 2020-02-18 at 3 04 52 PM" width="1164" src="https://user-images.githubusercontent.com/39598672/74774326-79fab400-5261-11ea-964d-1155fcbbd458.png">

@InnaAtanasova 
Aligned .

manu-kr commented:
> Not sure the dimensions should stay the same for Mobile. Right now you are using buttons in compact mode and not the cosy mode. For a11y requirements the buttons in mobile should be cosy, which will prob affect the height of the action bar. Could you please check this with the designers?
> 
> <img alt="Screen Shot 2020-02-18 at 3 30 39 PM" width="1318" src="https://user-images.githubusercontent.com/39598672/74775401-a1eb1700-5263-11ea-800a-41181391c0d9.png">

Sure i will check

stefanoScalzo commented:
> Instead of using media queries you can add modifier classes for the responsive behavior and let the user add these classes based on their needs. You can check the Page responsive behavior https://sap.github.io/fundamental-styles/layouts/page-layout.html

You removed the media queries but I don't seem to see the class modifiers?

manu-kr commented:
> > Instead of using media queries you can add modifier classes for the responsive behavior and let the user add these classes based on their needs. You can check the Page responsive behavior https://sap.github.io/fundamental-styles/layouts/page-layout.html
> 
> You removed the media queries but I don't seem to see the class modifiers?

@stefanoScalzo 

Now i have removed the media queries. and added modifier classes for responsive behaviour.

manu-kr commented:
> Instead of using media queries you can add modifier classes for the responsive behavior and let the user add these classes based on their needs. You can check the Page responsive behavior https://sap.github.io/fundamental-styles/layouts/page-layout.html

@InnaAtanasova 

I have removed media queries and added modifier classes for the responsive behaviour.

manu-kr commented:
> There's problem with the title alignment. The text is not centered vertically.
> <img alt="Screen Shot 2020-02-18 at 3 04 47 PM" width="1292" src="https://user-images.githubusercontent.com/39598672/74774325-79621d80-5261-11ea-81ee-01b3da50af98.png">
> <img alt="Screen Shot 2020-02-18 at 3 04 52 PM" width="1164" src="https://user-images.githubusercontent.com/39598672/74774326-79fab400-5261-11ea-964d-1155fcbbd458.png">

Fixed.

droshev commented:
@manu-kr can you fix the pipeline?

claassistantio commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/SAP/fundamental-styles?pullRequest=635) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/SAP/fundamental-styles?pullRequest=635) before we can accept your contribution.<br/>**7** out of **8** committers have signed the CLA.<br/><br/>:white_check_mark: JKMarkowski<br/>:white_check_mark: manu-kr<br/>:white_check_mark: mikerodonnell89<br/>:white_check_mark: salarenko<br/>:white_check_mark: droshev<br/>:white_check_mark: skvale<br/>:white_check_mark: InnaAtanasova<br/>:x: fundamental-bot<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/SAP/fundamental-styles?pullRequest=635) it.</sub>

manu-kr commented:
> @manu-kr can you fix the pipeline?

 Fixed the pipeline.

droshev commented:
@manu-kr your PR is touching 120 files, which is not normal. Also the PR is updating the images for components that you didn't touch.
Would you update the PR?

manu-kr commented:
This PR changes are moved to the PR #701 

https://github.com/SAP/fundamental-styles/pull/701

and closing this PR

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/SAP/fundamental-styles?pullRequest=635) <br/>All committers have signed the CLA.
